hspec-core/src/Test/Hspec/Core/Extension/Item.hs

```diff
@@ -0,0 +1,46 @@
+module Test.Hspec.Core.Extension.Item (
+-- * Types
+  Item(..)
+, Location(..)
+, Params(..)
+, ActionWith
+, Progress
+, ProgressCallback
+, Result(..)
+, ResultStatus(..)
+, FailureReason(..)
+
+-- * Operations
+, isFocused
+, pending
+, pendingWith
+, setAnnotation
+, getAnnotation
+) where
+
+import           Prelude ()
+import           Test.Hspec.Core.Compat
+
+import           Test.Hspec.Core.Spec hiding (pending, pendingWith)
+import           Test.Hspec.Core.Tree
+
+isFocused :: Item a -> Bool
```

Comment:
I think extensions should be able to modify every field of `Item`.

Reply:
I think they can do that already.  The `Item` type is exported.

Reply:
but the fields are not

Reply:
They are exported from `Test.Hspec.Core.Extension.Item`: https://github.com/hspec/hspec/pull/894/files#diff-edd9033d1fe072b0c2f04c047d3566881bbf49c63d961497a189d89854b9774bR3